### PR TITLE
Style selection: Implement Tracks events

### PR DESCRIPTION
**Pull request by fushar**
#### Proposed Changes

This PR adds event tracking around style selection flows. 821-gh-Automattic/dotcom-forge

#### Testing Instructions

1. On clicking a style selection button in the design picker

image

Verify that the following event is fired with the following props:

`calypso_signup_design_picker_style_variation_button_click` and the existing `calypso_signup_design_preview_select`

- `slug`: `<design slug>-<variation slug>`, e.g. `livro-white`
- `theme_style`:  `<theme>-<variation slug>`, e.g. `pub/livro-white`
- `has_style_variations`: `true`

---

2. On clicking a style selection preview in the design preview

image

Verify that the following event is fired:

`calypso_signup_design_preview_style_variation_preview_click`

with the same props as above:

- `slug`: `<design slug>-<variation slug>`, e.g. `livro-white`
- `theme_style`:  `<theme>-<variation slug>`, e.g. `pub/livro-white`
- `has_style_variations`: `true`

---

3. On clicking a device in the device switcher in the design preview

image

Verify that the following event is fired with the following props:

`calypso_signup_design_preview_device_click`

- `device`: `computer`/`tablet`/`phone`

---

4. On finally selecting a design

Verify that the following events: `calypso_signup_select_design` and `calypso_signup_design_type_submit`

has the following props:

- `slug`: `<design slug>-<variation slug>`, e.g. `livro-white`
- `theme_style`:  `<theme>-<variation slug>`, e.g. `pub/livro-white`
- `has_style_variations`: `true`

---

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to 821-gh-Automattic/dotcom-forge

### CODE REVIEW

**Comment by fushar** [comment]
cc: @autumnfjeld please do check about the naming convention.

I've deliberately prefixed the event names with either `design_picker` (not `design_card`) and `design_preview` depending on the screen. Let me know if this is acceptable!

**Comment by autumnfjeld** [comment]
@fushar  I will try to check this later today, or tomorrow. Thank you!!! 

FYI: This doesn't block the CfT. 😃

**Comment by taipeicoder** [comment]
Some feedback on my end, @autumnfjeld can make the call if we want to make adjustments based on them:

1. `calypso_signup_design_preview_select` doesn't have the prop `has_style_variations` for the old full-screen design preview.
Screen Shot 2022-09-19 at 5 05 15 PM

2. The slug `<design slug>-<variation slug>` only applies for style variations, for default style it is `<design slug>` and not `<design slug>-default`. So, for instance, the value will be `livro` instead of `livro-default`. The advantage of using `-default` prefix is that it can potentially be useful when doing Hue queries. I'm fine with either naming 🙂

**Comment by fushar** [comment]
> calypso_signup_design_preview_select doesn't have the prop has_style_variations for the old full-screen design preview.

Wow, nice catch! I've fixed it in the latest commit, thanks!

**Review by taipeicoder — APPROVED** [review]
Looking good on my end, let's wait on @autumnfjeld's feedback 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx
Comment (taipeicoder) [inline client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: Small detail, but for generated design preview `has_style_variations` will be `undefined` and thus the event won't have this prop. Not sure if it's a concern at the moment, that it'd be good to add it as `false` for consistency.

Comment (fushar) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: Nice find again 🤦 JS quirks. Fixed!

Comment (autumnfjeld) [reply client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/unified-design-picker.tsx]: Agreed. :)

**Comment by autumnfjeld** [comment]
> The slug <design slug>-<variation slug> only applies for style variations, for default style it is <design slug> and not <design slug>-default. So, for instance, the value will be livro instead of livro-default. 

I thought of this too, for different reasons, but it would break the continuity of looking at  just `livro`.   Although....hm...maybe that would be good, because it indicates a change in theme presentation, indicates a change in metrics meaning different things (e.g. now we have introduced style variations so metric analysis should be aware of that).  🤔

**Comment by autumnfjeld** [comment]
@fushar  Don't we need to update/add ` calypso_signup_select_design`

On finally selecting a design
` calypso_signup_select_design `

has the following prop:

slug: <design slug>-<variation slug>, e.g. livro-white
theme_style: <theme>-<variation slug>, e.g. pub/livro-white
has_style_variations: true


(Actually maybe you did this in the code, just the description above didn't reflect that.)

**Comment by fushar** [comment]
> (Actually maybe you did this in the code, just the description above didn't reflect that.)

This is what happens 😄

**Comment by fushar** [comment]
> > The slug - only applies for style variations, for default style it is  and not -default. So, for instance, the value will be livro instead of livro-default.
> 
> I thought of this too, for different reasons, but it would break the continuity of looking at just `livro`. Although....hm...maybe that would be good, because it indicates a change in theme presentation, indicates a change in metrics meaning different things (e.g. now we have introduced style variations so metric analysis should be aware of that). 🤔

Do we want to do this? I'm perfectly fine either way, no preferences.

**Comment by autumnfjeld** [comment]
> Do we want to do this? I'm perfectly fine either way, no preferences.

Let's leave it as is. :)

**Review by autumnfjeld — APPROVED** [review]
Once this is merged , after a day or two I'll do another review to make sure the events are firing as expected. :)

**Pull request merged**
